internal/gps/result.go

```diff
@@ -61,19 +65,45 @@ func WriteDepTree(basedir string, l Lock, sm SourceManager, sv bool) error {
 		return err
 	}
 
-	// TODO(sdboyer) parallelize
-	for _, p := range l.Projects() {
-		to := filepath.FromSlash(filepath.Join(basedir, string(p.Ident().ProjectRoot)))
+	projects := l.Projects()
+	worksz := len(projects)
+
+	workChan := make(chan LockedProject, worksz)
```

Comment:
so, my default position is that there are only two correct values for channel sizes: 0, and 1. while we could do it this way, it's creating problems elsewhere (like early termination, noted below). i think a refactored implementation that uses unbuffered channels would give us tighter control over such things.